### PR TITLE
Updated Musee Middleware Logic

### PR DESCRIPTION
**Pull request by xmas7**
Updated Musee Middleware Logic

### CODE REVIEW

File: MUSEEMiddleware.sol [hunk 1]
```diff
@@ -379,6 +379,40 @@ contract MUSEEMiddleware is Constants {
...
+      if (address(creatorRecipients[i]).isContract()) {
+        try PercentSplitETH(creatorRecipients[i]).getShareLength{ gas: READ_ONLY_GAS_LIMIT }() returns (
+          uint256 recipientCount
+        ) {
+          uint256 totalSplitShares;
+          for (uint256 splitIndex = 0; splitIndex < recipientCount; ++splitIndex) {
+            uint256 share = PercentSplitETH(creatorRecipients[i]).getPercentInBasisPointsByIndex(splitIndex);
+            totalSplitShares += share;
+          }
+          for (uint256 splitIndex = 0; splitIndex < recipientCount; ++splitIndex) {
```
Comment (wlm84) [inline MUSEEMiddleware.sol:395]: amazing

File: MUSEEMiddleware.sol [hunk 2]
```diff
@@ -379,6 +379,40 @@ contract MUSEEMiddleware is Constants {
...
+    
+    // Populate rev splits, including any percent splits
+    uint256 revSplitIndex = 0;
+    for (uint256 i = 0; i < creatorRecipients.length; ++i) {
+      if (address(creatorRecipients[i]).isContract()) {
+        try PercentSplitETH(creatorRecipients[i]).getShareLength{ gas: READ_ONLY_GAS_LIMIT }() returns (
+          uint256 recipientCount
+        ) {
+          uint256 totalSplitShares;
+          for (uint256 splitIndex = 0; splitIndex < recipientCount; ++splitIndex) {
```
Comment (wlm84) [inline MUSEEMiddleware.sol:391]: sure its less than Count!

File: MUSEEMiddleware.sol [hunk 3]
```diff
@@ -379,6 +379,40 @@ contract MUSEEMiddleware is Constants {
       // Nothing
       details = string.concat(details, " has not listed nor gotten an offer");
     }
+    
+    // Populate rev splits, including any percent splits
+    uint256 revSplitIndex = 0;
+    for (uint256 i = 0; i < creatorRecipients.length; ++i) {
```
Comment (wlm84) [inline MUSEEMiddleware.sol:385]: what if length returns null?

**Review by wlm84 — COMMENTED** [review]
awesome, going real quick

File: MUSEEMiddleware.sol (hunk 2 above)
Comment (wlm84) [reply MUSEEMiddleware.sol:391]: must be less than it

Comment (xmas7) [reply MUSEEMiddleware.sol:391]: absolutely

File: MUSEEMiddleware.sol (hunk 3 above)
Comment (xmas7) [reply MUSEEMiddleware.sol:385]: it can't be null

**Pull request merged**